### PR TITLE
Fw 5069 Refactored hydration utility and added more tests for search related utilities and views

### PR DESCRIPTION
**Pull request by sahil97**
### Description of Changes
1. Refactored hydration method to use minimal serializers for all responses, thus not requiring request be passed down.
2. Added PersonMinimalSerializer
3. Added tests for 
  - hydration of all objects, i.e. dictionary_entries, songs, stories, images, audio and video
  - testing out errors during hydration process
  - error cases in base search view
4. Excluded search/tasks folder, added that info to a separate ticket to cover that when doing either integration testing or some other way to test celery-elasticsearch interaction.
5. Coverage should be up to ~85% now.

### Checklist
- ~README / documentation has been updated if needed~
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- ~Admin Panel has been updated if models have been added or modified~
- ~Migrations have been updated and committed if applicable~
- ~Team Postman workspace has been updated if applicable~

### Reviewers Should Do The Following:
- [x] Review the code changes here
- ~Pull the branch and test locally~

### Additional Notes
N/A

### CODE REVIEW

File: firstvoices/backend/tests/test_apis/test_search_api.py [hunk 1]
```diff
@@ -181,3 +183,132 @@ def test_invalid_page_sizes(self, page_size, db, mock_search_query_execute):
...
+            }
+        }
+
+        mock_search_query_execute.return_value = mock_es_results
+
+        response = self.client.get(
+            self.get_list_endpoint(site_slug=site.slug) + "?startsWithChar=x"
+        )
+
+        assert "startsWithChar=x" in response.request["QUERY_STRING"]
```
Comment (sarahfirstvoices) [inline firstvoices/backend/tests/test_apis/test_search_api.py:314]: I don't think this is testing anything useful for us (just that you used the correct url on line 311). A better assertion would be something like 

`mock_search_query_execute.assert_called_with(my_expected_value)`

**Review by sarahfirstvoices — CHANGES_REQUESTED** [review]
Mostly looking great! Just that one issue about the startsWithChar test

**Review by bronson-c — APPROVED** [review]
LGTM outside of the one test that Sarah has requested changes on, great stuff 👍

File: firstvoices/backend/tests/test_apis/test_search_api.py [hunk 2]
```diff
@@ -278,37 +278,19 @@ def test_invalid_category_id(self):
...
-
-        mock_search_query_execute.return_value = mock_es_results
-
-        response = self.client.get(
+        self.client.get(
             self.get_list_endpoint(site_slug=site.slug) + "?startsWithChar=x"
         )
 
-        assert "startsWithChar=x" in response.request["QUERY_STRING"]
+        assert mock_get_search_query.call_args.kwargs["starts_with_char"] == "x"
```
Comment (sarahfirstvoices) [inline firstvoices/backend/tests/test_apis/test_search_api.py:296]: I love this!

**Review by sarahfirstvoices — APPROVED** [review]

**Pull request merged**